### ISSUE TITLE
tx-error page times out

### ISSUE THREAD

**Issue by grooviegermanikus** [blockworks-foundation/solana-bankingstage-dashboard#49]
This query is slow
```sql
 SELECT * FROM (
	SELECT
		signature,
		txi.transaction_id,
		( SELECT count(distinct slot) FROM banking_stage_results_2.transaction_slot WHERE transaction_id=tx_slot.transaction_id ) AS num_relative_slots,
		(
		   SELECT ARRAY_AGG(json_build_object('slot', tx_slot.slot, 'error', err.error_text, 'count', count)::text)
		   FROM banking_stage_results_2.errors err
		   WHERE err.error_code=tx_slot.error_code
	   ) AS all_errors,
	   tx_slot.utc_timestamp,
	   -- optional fields from transaction_infos
	   ( txi is not null ) AS was_included_in_block,
	   txi.cu_requested,
	   txi.prioritization_fees
	FROM banking_stage_results_2.transactions txs
	LEFT JOIN banking_stage_results_2.transaction_slot tx_slot ON tx_slot.transaction_id=txs.transaction_id
	LEFT JOIN banking_stage_results_2.transaction_infos txi ON txi.transaction_id=tx_slot.transaction_id
	WHERE true
	 -- in list mode:
		AND tx_slot IS NOT NULL
		--AND (%s or signature = %s)
) AS data
-- transaction_id is required as tie breaker
ORDER BY utc_timestamp DESC, transaction_id DESC
```

.. compared to 
```sql
 SELECT * FROM (
	SELECT
		signature,
		txi.transaction_id,
		( SELECT count(distinct slot) FROM banking_stage_results_2.transaction_slot WHERE transaction_id=tx_slot.transaction_id ) AS num_relative_slots,
		(
		   SELECT ARRAY_AGG(json_build_object('slot', tx_slot.slot, 'error', err.error_text, 'count', count)::text)
		   FROM banking_stage_results_2.errors err
		   WHERE err.error_code=tx_slot.error_code
	   ) AS all_errors,
	   tx_slot.utc_timestamp,
	   -- optional fields from transaction_infos
	   ( txi is not null ) AS was_included_in_block,
	   txi.cu_requested,
	   txi.prioritization_fees
	FROM banking_stage_results_2.transactions txs
	INNER JOIN banking_stage_results_2.transaction_slot tx_slot ON tx_slot.transaction_id=txs.transaction_id
	LEFT JOIN banking_stage_results_2.transaction_infos txi ON txi.transaction_id=tx_slot.transaction_id
	WHERE true
		--AND (%s or signature = %s)
) AS data
-- transaction_id is required as tie breaker
ORDER BY utc_timestamp DESC, transaction_id DESC
```

**Comment by grooviegermanikus** [comment]
See roughly this discussion https://stackoverflow.com/questions/24876673/explain-join-vs-left-join-and-where-condition-performance-suggestion-in-more-de

tl;dr: sometimes the combination of LEFT JOIN and `WHERE` does not perform as well as `LEFT JOIN`

**Comment by grooviegermanikus** [comment]
fixed by dynamically switching INNER JOIN + LEFT JOIN

**Comment by grooviegermanikus** [comment]
note, if signature is provided (by search) we want to return if either tx_slot or tx_info matches:
`(%s or signature = %s AND (tx_slot IS NOT NULL OR txi IS NOT NULL))`

whereas in the tx-error list we want transactions only if tx_slot is set

**Comment by grooviegermanikus** [comment]
deployed
